### PR TITLE
feat: `noNativeElements` renders table with flex layout

### PR DESCRIPTION
**Pull request by ling1726**
Native `display: table` is great for column alignment and simplifies column sizing. However, native table layout is not very virtualization friendly and does cannot handle any non-compliant element in the markup.

For virtualization and other cases that involve non-compliant table elements in markup, `noNativeElements` now renders the table also in a flex layout.

Flex layout is what was used before #24762 with fixes to column alignment for long content.

The recommended usage of table should be native semantic elements. This is because of:
* Accessibility - native `table` elements are the most accessible
* Performance - `display: table` is simply faster at rendering tables than `display; flex` [we did benchmarks on this here](https://github.com/ling1726/table-layout-perf).

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/3f105cca41d951f161edbb90be116565311e72d1'>3f105cca41d951f161edbb90be116565311e72d1</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24913/perf-test-react-components/Avatar-mount_base.html">1317</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24913/perf-test-react-components/Avatar-mount.html">1317</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24913/perf-test-react-components/Button-mount_base.html">955</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24913/perf-test-react-components/Button-mount.html">946</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24913/perf-test-react-components/FluentProvider-mount_base.html">1601</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24913/perf-test-react-components/FluentProvider-mount.html">1581</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24913/perf-test-react-components/FluentProviderWithTheme-mount_base.html">631</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24913/perf-test-react-components/FluentProviderWithTheme-mount.html">631</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24913/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">593</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24913/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">604</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24913/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">635</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24913/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">634</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24913/perf-test-react-components/MakeStyles-mount_base.html">1894</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24913/perf-test-react-components/MakeStyles-mount.html">1906</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24913/perf-test-react-components/SpinButton-mount_base.html">2496</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24913/perf-test-react-components/SpinButton-mount.html">2512</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Review by bsunderhus — APPROVED** [review]
LGTM

File: packages/react-components/react-table/src/components/Table/useTableStyles.ts [hunk 1]
```diff
@@ -26,7 +38,16 @@ const useStyles = makeStyles({
  */
 export const useTableStyles_unstable = (state: TableState): TableState => {
   const styles = useStyles();
-  state.root.className = mergeClasses(tableClassName, styles.root, state.root.className);
+  const layoutStyles = {
+    table: useTableLayoutStyles(),
+    flex: useFlexLayoutStyles(),
+  };
```
Comment (layershifter) [inline packages/react-components/react-table/src/components/Table/useTableStyles.ts:44]: nit: not sure that I understand the need to create an object there

File: packages/react-components/react-table/src/components/TableCell/useTableCellStyles.ts [hunk 2]
```diff
@@ -8,14 +8,28 @@ export const tableCellClassNames: SlotClassNames<TableCellSlots> = {
...
+});
+
+const useFlexLayoutStyles = makeStyles({
+  root: {
+    display: 'flex',
+    minWidth: '0px',
+    alignItems: 'center',
+    ...shorthands.flex(1, 1, '0px'),
+  },
+});
```
Comment (layershifter) [inline packages/react-components/react-table/src/components/TableCell/useTableCellStyles.ts:25]: Can it be just:

```ts
const useLayoutStyles = makeStyles({
  table: {
    display: 'table-cell',
    verticalAlign: 'middle',
  },
  flex: {
    display: 'flex',
    minWidth: '0px',
    alignItems: 'center',
    ...shorthands.flex(1, 1, '0px'),
  },
});
```

**Pull request merged**